spec/services/activity/import_spec.rb

```diff
@@ -1266,13 +1264,25 @@
           "Tags" => "",
           "Original commitment figure" => "100000"
         }
+      }
+
+      before { allow(ActivityPolicy).to receive(:new).with(uploader, ispf).and_return(activity_policy_double) }
 
+      it "prefixes the RODA identifier with 'NODA'" do
         expect { subject.import([new_non_oda_programme_attributes]) }.to change { Activity.count }.by(1)
 
         new_activity = Activity.order(:created_at).last
 
         expect(new_activity.roda_identifier).to start_with("NODA-")
       end
+
+      it "sets publish_to_iati to false" do
+        expect { subject.import([new_non_oda_programme_attributes]) }.to change { Activity.count }.by(1)
```

Comment:
I know it's in the test above for some reason, but is there a reason we need to check that the import creates a new activity in this test?

Reply:
Good question. I suppose it gives us confidence that when we set the `new_activity` variable, we're actually checking a new activity rather than an existing one?

Reply:
I'm actually now wondering if this test is necessary in addition to the other tests we have. We currently have:
1. Unit test in the activity class - this feels like the most important one, as it's where the actual behaviour is written.
2. Feature tests that test it's set after the activity form
3. Service (semi-integration-style) tests here that test that it's created with the field in the importer. There's not actually any specific behaviour in the importer that sets this field value, unless I've missed something, so to me testing this value here potentially feels outside the responsibility of this class.

I wonder if it's worth testing 1 and 2 (potentially adding a spec to the `spec/features/users_can_upload_activities_spec.rb` and the BEIS alternative if we feel it needs more coverage).

I'm probably over thinking things, but what do you think?

Reply:
Yeah, Joe and I were talking about this before. We noticed that there's no equivalent to 2 for the importer in `spec/features/users_can_upload_activities_spec.rb` and the BEIS version, so I added it in here. The upload/import feature specs don't seem to test that all the values are right in the same way as the form/UI feature specs, but maybe they should (or we could at least move this and the previous two tests over there?)?

The `prefixes the RODA identifier with 'NODA` test is one you wrote actually, but I think the same applies to that: both the testing that an activity is created and that it's not testing functionality of the importer itself, just that functionality also applies when using the importer

Reply:
Yeah, I think it would be good to try and reduce the scope of the importer's tests if we can move this kind of thing to the feature tests, although I appreciate this feels like a bit of scope creep in terms of what this PR is supposed to be doing.

I think both of those tests could be moved!

Reply:
I think to keep scope down, we can park that stuff where it is for now and add it as something to refactor/consider when we get the time to refactor the importer spec?

Reply:
Sounds good - shall I merge for now and add a note on the import spec refactor card then?

Reply:
Or did you mean just the testing of all fields should be parked?

Reply:
Let's merge now and update that card with this new plan